### PR TITLE
AP_Param: greatly speed up param download with disabled parameters

### PR DESCRIPTION
**Pull request by tridge**
allows fast skip over disabled subgroups. This removes a long delay
with param download on a MatekF405-STD where the final parameters
associated with the OSD took 80ms to fetch, causing a long loop delay
Thanks to the bug report by @giacomo892 
This also fixes the display of disabled ENABLE vars for INS_TCALn_ENABLE

### CODE REVIEW

File: libraries/AP_Param/AP_Param.h [hunk 1]
```diff
@@ -210,8 +210,9 @@ class AP_Param
     // a token used for first()/next() state
     typedef struct {
         uint32_t key : 9;
-        uint32_t idx : 5; // offset into array types
+        uint32_t idx : 4; // offset into array types
```
Comment (davidbuzz) [inline libraries/AP_Param/AP_Param.h:213]: I'm not super familiar with the code, but from the outside, it looks like ParamToken.idx used to hold 5bits of data and now only holds 4, is that a safe thing to do?   You've also moved the bit position of group_element down by one,  again, I'm not sure of the risk ..but I imagine a scenario where a ParamToken's 32bits are filled bitwise from a flash that was written bitwise with the old code, and end up bits doing things they shouldn't...?

**Review by davidbuzz — APPROVED** [review]
Had a read thru all of this, and it seems to align with the discussion we had b4 u wrote it. I haven't tested it, but I assume u have. :-)

File: libraries/AP_Param/AP_Param.h (hunk 1 above)
Comment (tridge) [reply libraries/AP_Param/AP_Param.h:213]: it actually only needs 2 bits. It is used to index into parameter arrays, and the only param arrays we have are AP_Vector3f, which index from 0 to 2.
We had it as 5 bits to make it all neatly 32 bits total.
The ordering is needed so I didn't need to change this table:
https://github.com/ardupilot/ardupilot/blob/master/libraries/AP_OSD/AP_OSD_ParamScreen.cpp#L130

Comment (tridge) [reply libraries/AP_Param/AP_Param.h:213]: also note that ParamToken is in memory only, it isn't a persistent data structure

**Comment by rmackay9** [comment]
We suffered a bit not having more people qualified in this area to reviews the PR but it's been tested so hopefully it will be OK.

**Pull request merged**